mempool/v0/clist_mempool.go

```diff
@@ -613,6 +614,15 @@ func (mem *CListMempool) Update(
 		// https://github.com/tendermint/tendermint/issues/3322.
 		if e, ok := mem.txsMap.Load(tx.Key()); ok {
 			mem.removeTx(tx, e.(*clist.CElement), false)
+			// see if the transaction is a malleated transaction of a some parent
+			// transaction that exists in the mempool
+		} else if malleatedTx, isMalleated := types.UnwrapMalleatedTx(tx); isMalleated {
```

Comment:
Coming back to this...
`PrepareProposal` (at least in the version shipped in `v0.37.x`) will not touch the mempool when the application gives back its set of transactions (in Celestia's case, some of them malleated). Therefore I don't see why we would find malleated transactions (AFAIU these are created at `PrepareProposal`) in the mempool.

Reply:
OK, I came back to this after our discussion today and I confirm I had misunderstood these changes.
We indeed look into the mempool data structures, but only to remove the TX of the current loop iteration, which is a transaction decided in a block (and therefore can be malleated).
Still, the discussion we have on doing this in re-check TX is still relevant, but I guess we can resolved this comment.